### PR TITLE
feat(matching): parallel ASR transcription + honest dashboard concurrency

### PR DESCRIPTION
**Pull request by Jsakkos**
## Summary

Fixes the "many tracks show **Matching** but only one actually progresses (and the CPU looks idle)" behavior. This was **not** a UI bug and **not** a stuck semaphore: the shared Whisper model was built with faster-whisper's default `num_workers=1`, so concurrent transcriptions serialized behind a single worker while the admission semaphore — sized from `max_concurrent_matches` — kept admitting many tracks into `MATCHING`. The semaphore (admission) and the model's real concurrency (1) were decoupled, so the count overstated progress.

**Piece A** makes ASR genuinely parallel and the dashboard honest:

- The shared model now runs **N parallel workers** (faster-whisper `num_workers`, sharing one set of weights — no per-worker weight-memory multiplication), sized from `max_concurrent_matches` and **hardware-clamped**: CPU clamps to physical cores with `cpu_threads = cores // workers` (avoids the oversubscription that makes naive parallelism *slower*); GPU clamps to a conservative cap of 4.
- The **match semaphore is tied to that real worker capacity**, so exactly `workers` tracks show `MATCHING` and the rest correctly show `QUEUED` — the count can no longer overstate progress.
- A new **`GET /api/asr-status`** endpoint + a small dashboard **ASR badge** surface the active backend (e.g. `ASR: CUDA · float16 · 4w` / `ASR: CPU · int8 · 8w`).
- The **ConfigWizard hint is now honest** (hardware-clamped, takes effect after a backend restart) and the input bound is consistent.

GPU acceleration in the shipped release binary (bundling the CUDA libraries) is a separate, heavier effort tracked as **Piece B** and is intentionally out of scope here. ASR parallelism benefits the GPU-from-source path today too, since `num_workers=1` previously serialized GPU transcriptions as well.

Single source of truth: `resolve_asr_runtime(device, requested_workers)` in `asr_models.py` is consumed by the model loader, the semaphore, and the status endpoint, so all three always agree.

## Changes

- `asr_models.py` — `AsrRuntime` / `detect_asr_device` / `resolve_asr_runtime`; `FasterWhisperModel` passes `num_workers`/`cpu_threads`; both `_model_cache` keys made worker-aware (no stale single-worker reuse).
- `episode_identification.py` — `EpisodeMatcher` carries `requested_workers` via a DRY `_model_config()` helper.
- `curator.py` — forwards `config.max_concurrent_matches` as the matcher's `requested_workers`.
- `job_manager.py` — sizes the match semaphore from `resolve_asr_runtime(...).workers`.
- `routes.py` — `GET /api/asr-status`.
- Frontend — `AsrStatusBadge.tsx` (mounted in `App.tsx`); ConfigWizard hint/bound.

## Test Plan

- [x] Backend: **1751** unit + matcher tests pass; `ruff check app/ tests/` clean
- [x] Frontend: `npm run build` (tsc + vite) passes
- [x] New unit coverage: runtime sizing math (CPU thread-division, GPU cap, edge cases), model `num_workers`/`cpu_threads` kwargs + worker-aware cache keys (incl. GPU omits `cpu_threads`), semaphore sized to resolved workers, `/api/asr-status` fields
- [ ] Manual (GPU-from-source): on a multi-track disc, confirm exactly *N* tracks show **MATCHING** and the rest **QUEUED**, multiple transcriptions progress concurrently, and the ASR badge reflects the active device/worker count

Spec & plan: `docs/superpowers/specs/2026-06-04-parallel-asr-matching-visibility.md`, `docs/superpowers/plans/2026-06-04-parallel-asr-matching-visibility.md`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/app/matcher/asr_models.py [hunk 1]
```diff
@@ -406,7 +477,12 @@ def get_cached_model(model_config: dict) -> ASRModel:
...
     """
-    cache_key = f"{model_config.get('type', '')}_{model_config.get('name', '')}_{model_config.get('device', 'auto')}"
+    device = model_config.get("device") or detect_asr_device()
+    runtime = resolve_asr_runtime(device, model_config.get("requested_workers", 1))
+    cache_key = (
+        f"{model_config.get('type', '')}_{model_config.get('name', '')}"
+        f"_{device}_w{runtime.workers}_t{runtime.cpu_threads}"
+    )
 
     if cache_key not in _model_cache:
```
[the remark this answers is not in the document]
Comment (Jsakkos) [reply backend/app/matcher/asr_models.py]: Good catch — fixed in b9c026b. `get_cached_model`'s check-create-store is now wrapped in a module-level `threading.Lock`, so exactly one shared model is built per `cache_key`; the other N-1 threads block on the lock, then reuse it. First-load contention is intentional here — it's what makes "one shared model with N workers" actually *one* model. Added a regression test (`test_concurrent_get_cached_model_constructs_once`) that races 8 threads on a cold cache through a barrier and asserts a single construction.

File: backend/app/api/routes.py [hunk 2]
```diff
@@ -3594,6 +3594,24 @@ async def get_update_status():
...
+    """Resolved ASR backend for the dashboard badge (read-only, no secrets)."""
+    from app.matcher.asr_models import detect_asr_device, resolve_asr_runtime
+    from app.services.config_service import get_config as _get_app_config
+
+    config = await _get_app_config()
+    runtime = resolve_asr_runtime(detect_asr_device(), config.max_concurrent_matches)
+    return {
+        "device": runtime.device,
+        "compute_type": runtime.compute_type,
+        "model": "small",  # current hardcoded matcher default (EpisodeMatcher.model_name)
```
[the remark this answers is not in the document]
Comment (Jsakkos) [reply backend/app/api/routes.py:3608]: The literal suggestion would raise `AttributeError`: `model_name` is an *instance* attribute set in `EpisodeMatcher.__init__` (`self.model_name = model_name`), not a class attribute, so `EpisodeMatcher.model_name` isn't accessible without an instance. Model size isn't user-configurable yet (a deferred non-goal of this PR), so `"small"` is the genuine value across the matcher; I'm keeping the accurately-commented hardcode for v1. When model size does become configurable, the right move is the shared module-level constant you suggested as the alternative — I'll wire that in then so the badge and the matcher default stay in sync.

File: backend/app/api/routes.py [hunk 3]
```diff
@@ -3594,6 +3594,24 @@ async def get_update_status():
...
 
 
+@router.get("/asr-status")
+async def get_asr_status():
+    """Resolved ASR backend for the dashboard badge (read-only, no secrets)."""
+    from app.matcher.asr_models import detect_asr_device, resolve_asr_runtime
+    from app.services.config_service import get_config as _get_app_config
+
+    config = await _get_app_config()
+    runtime = resolve_asr_runtime(detect_asr_device(), config.max_concurrent_matches)
```
[the remark this answers is not in the document]
Comment (Jsakkos) [reply backend/app/api/routes.py:3604]: Real edge case, acknowledged — tracking it as a known v1 limitation rather than fixing here. Worth noting it also affects the startup-sized semaphore (sized from `detect_asr_device()` before any model loads), so a complete fix is the post-load-truth approach you describe applied to all three consumers, which interacts with startup ordering and the deferred Piece B "GPU in the release binary" work. On the supported setups it's accurate today: GPU-from-source has the CUDA libraries present, and CPU-only frozen builds report 0 CUDA devices from ctranslate2. The badge is best-effort/cosmetic, so a transient device mismatch in the rare "CUDA visible but libraries missing" case can't affect matching itself.
